crates/sui-core/src/safe_client.rs

```diff
@@ -310,6 +310,27 @@ impl<C> SafeClient<C>
 where
     C: AuthorityAPI + Send + Sync + Clone + 'static,
 {
+    /// Submit a transaction for certification and execution.
+    pub async fn submit_transaction(
+        &self,
+        request: SubmitTransactionRequest,
+        client_addr: Option<SocketAddr>,
+    ) -> Result<SubmitTransactionResponse, SuiError> {
+        let _timer = self.metrics.handle_certificate_latency.start_timer();
+        let response = self
+            .authority_client
+            .submit_transaction(request, client_addr)
+            .await?;
+
+        // TODO(fastpath): verify response.
+        // let verified = check_error!(
+        //     self.address,
+        //     self.verify_certificate_response_v3(&digest, response),
+        //     "Client error in handle_certificate"
+        // )?;
```

Comment:
Should probably handle this TODO before merging, shouldn't be much code

Reply:
I think this would be basically `verify_certificate_response_v3()` but minus `check_signed_effects_plain()`.